circuits/cpp/barretenberg/cpp/src/barretenberg/proof_system/relations/ultra_relation_consistency.test.cpp

```diff
@@ -0,0 +1,534 @@
+/**
+ * @file ultra_relation_consistency.test.cpp
+ * @brief Show that relation arithmetic has a simple form.
+ * @details The purpose of this test suite is to show that the identity arithmetic implemented in the Relations is
+ * equivalent to a simpler unoptimized version implemented in the tests themselves. This is useful 1) as documentation
+ * since the simple implementations here should make the underlying arithmetic easier to see, and 2) as a check that
+ * optimizations introduced into the Relations have not changed the result.
+ *
+ * For this purpose, we simply feed (the same) random inputs into each of the two implementations and confirm that
+ * the outputs match. This does not confirm the correctness of the identity arithmetic (the identities will not be
+ * satisfied in general by random inputs) only that the two implementations are equivalent.
+ *
+ */
+#include "barretenberg/ecc/curves/bn254/fr.hpp"
+#include "barretenberg/proof_system/relations/auxiliary_relation.hpp"
+#include "barretenberg/proof_system/relations/elliptic_relation.hpp"
+#include "barretenberg/proof_system/relations/gen_perm_sort_relation.hpp"
+#include "barretenberg/proof_system/relations/lookup_relation.hpp"
+#include "barretenberg/proof_system/relations/permutation_relation.hpp"
+#include "barretenberg/proof_system/relations/ultra_arithmetic_relation.hpp"
+#include <gtest/gtest.h>
+
+using namespace proof_system;
+
+namespace proof_system::ultra_relation_consistency_tests {
+
+using FF = barretenberg::fr;
+struct InputElements {
+    static constexpr size_t NUM_ELEMENTS = 43;
+    std::array<FF, NUM_ELEMENTS> _data;
+
+    static InputElements get_random()
+    {
+        InputElements result;
+        std::generate(result._data.begin(), result._data.end(), [] { return FF::random_element(); });
+        return result;
+    }
+
+    static InputElements get_special()
```

Comment:
can you add a comment explaining why this is special?

Reply:
done